sources/packages/api/src/services/institution-location/institution-location.service.ts

```diff
@@ -56,6 +57,57 @@ export class InstitutionLocationService extends RecordDataModelService<Instituti
     return this.repo.save(saveLocation);
   }
 
+  /**
+   * Updating institution location.
+   * @param data Payload of updated data.
+   * @param locationId Location Id to update.
+   * @returns Updated location as InstitutionLocation.
+   */
+  async updateLocationPrimaryContact(
+    data: Partial<InstitutionLocationPrimaryContactModel>,
+    locationId?: number,
+  ): Promise<InstitutionLocation> {
+    const saveLocation = {
+      primaryContact: {
+        firstName: data.primaryContactFirstName,
+        lastName: data.primaryContactLastName,
+        email: data.primaryContactEmail,
+        phone: data.primaryContactPhone,
+      },
+      id: locationId ?? undefined,
+    };
+
+    return this.repo.save(saveLocation);
+  }
+
+  /**
+   * Updating institution location.
+   * @param data Payload of updated data.
+   * @param locationId Location Id to update.
+   * @returns Updated location as InstitutionLocation.
+   */
+  async updateLocationForAEST(
+    data: InstitutionLocationModel,
+    locationId?: number,
+  ): Promise<InstitutionLocation> {
+    const saveLocation = {
+      name: data.locationName,
+      data: {
+        address: transformAddressDetails(data),
+      },
+      primaryContact: {
+        firstName: data.primaryContactFirstName,
+        lastName: data.primaryContactLastName,
+        email: data.primaryContactEmail,
+        phone: data.primaryContactPhone,
+      },
+      institutionCode: data.institutionCode,
+      id: locationId ?? undefined,
```

Comment:
is this function supposed to create a new `InstitutionLocation` if the `id` is not present?